src/gentropy/dataset/study_locus.py

```diff
@@ -628,3 +628,68 @@ def _qc_no_population(self: StudyLocus) -> StudyLocus:
             ),
         )
         return self
+
+    def annotate_locus_statistics_boundaries(
+        self: StudyLocus,
+        summary_statistics: SummaryStatistics,
+    ) -> StudyLocus:
+        """Annotates study locus with summary statistics in the specified boundaries - locusStart and locusEnd.
+
+        Args:
+            summary_statistics (SummaryStatistics): Summary statistics to be used for annotation.
+
+        Returns:
+            StudyLocus: Study locus annotated with summary statistics in `locus` column. If no statistics are found, the `locus` column will be empty.
+        """
+        # The clumps will be used several times (persisting)
+        self.df.persist()
+        # Renaming columns:
+        sumstats_renamed = summary_statistics.df.selectExpr(
+            *[f"{col} as tag_{col}" for col in summary_statistics.df.columns]
+        ).alias("sumstat")
+
+        locus_df = (
+            sumstats_renamed
+            # Joining the two datasets together:
+            .join(
+                f.broadcast(
+                    self.df.alias("clumped").select(
+                        "position",
+                        "chromosome",
+                        "studyId",
+                        "studyLocusId",
+                        "locusStart",
+                        "locusEnd",
+                    )
+                ),
+                on=[
+                    (f.col("sumstat.tag_studyId") == f.col("clumped.studyId"))
+                    & (f.col("sumstat.tag_chromosome") == f.col("clumped.chromosome"))
+                    & (f.col("sumstat.tag_position") >= (f.col("clumped.locusStart")))
+                    & (f.col("sumstat.tag_position") <= (f.col("clumped.locusEnd")))
+                ],
+                how="inner",
+            )
+            .withColumn(
+                "locus",
+                f.struct(
+                    f.col("tag_variantId").alias("variantId"),
+                    f.col("tag_beta").alias("beta"),
+                    f.col("tag_pValueMantissa").alias("pValueMantissa"),
+                    f.col("tag_pValueExponent").alias("pValueExponent"),
+                    f.col("tag_standardError").alias("standardError"),
+                ),
+            )
+            .groupBy("studyLocusId")
+            .agg(
+                f.collect_list(f.col("locus")).alias("locus"),
+            )
+        )
+
+        self.df = self.df.drop("locus").join(
```

Comment:
At the end we are missing out some columns from the `StudyLocus` schema - 
actual schema: 
```
 |-- locus: array (nullable = true)
 |    |-- element: struct (containsNull = false)
 |    |    |-- variantId: string (nullable = false)
 |    |    |-- beta: double (nullable = false)
 |    |    |-- pValueMantissa: float (nullable = false)
 |    |    |-- pValueExponent: integer (nullable = false)
 |    |    |-- standardError: double (nullable = true)
```

vs the one we expect in the object:
```
|-- locus: array (nullable = true)
 |    |-- element: struct (containsNull = true)
 |    |    |-- is95CredibleSet: boolean (nullable = true)
 |    |    |-- is99CredibleSet: boolean (nullable = true)
 |    |    |-- logBF: double (nullable = true)
 |    |    |-- posteriorProbability: double (nullable = true)
 |    |    |-- variantId: string (nullable = true)
 |    |    |-- pValueMantissa: float (nullable = true)
 |    |    |-- pValueExponent: integer (nullable = true)
 |    |    |-- beta: double (nullable = true)
 |    |    |-- standardError: double (nullable = true)
 |    |    |-- r2Overall: double (nullable = true)
```

Reply:
This is because the second part is credible set that shouldn't be there for a locus collection. The actual schema represents GWAS sumstats collected in the locus.

Reply:
The validation code overwrites the schema with null values just for nullable columns, so we are safe here and there is no error - https://github.com/opentargets/gentropy/blob/6d931926228a099ac98762a290b1af475da01c11/src/gentropy/dataset/dataset.py#L136C72-L136C78